script.js

```diff
@@ -29,4 +39,100 @@ function operate(operator, a, b) {
         case "/":
             return divide(a,b);
     }
-}
\ No newline at end of file
+}
+
+function updateInputDisplay(list) {
+    displayScreen = document.querySelector(`.input`);
+    displayScreen.textContent = list.join("");
+}
+
+function updateResultDisplay(value) {
+    displayScreen = document.querySelector(`.results`);
+    displayScreen.textContent = value;
+}
+
+function clearAll() {
+    characters = [];
+    operator = [];
+    inputsForDisplay = [];
+    reversePolishNotation = [];
+    runningResult = 0;
+    updateInputDisplay([])
+    updateResultDisplay("0")
+}
+
+const clearButton = document.querySelector(".clear");
+clearButton.addEventListener("click", (e) => {
+    clearAll()
+    console.log(characters, reversePolishNotation, operator);
+})
+
+const digitButtons = document.querySelectorAll(".digit");
+digitButtons.forEach( (digitButton) => {
+    digitButton.addEventListener("click", (e) => {
+        digit = e.target.getAttribute("data-value")
+        if (OPERATORS.includes(characters.toString())) {
+            characters = [digit]
+            inputsForDisplay.push(digit);
+            updateInputDisplay(inputsForDisplay);
+            console.log(characters, reversePolishNotation, operator);
+        } else if (characters.length == 0 || NUMBERS.includes(characters[characters.length-1])) {
+            characters.push(digit);
+            inputsForDisplay.push(digit);
+            updateInputDisplay(inputsForDisplay);
+            console.log(characters, reversePolishNotation, operator);
+        }
+
+    })
+})
+
+const operatorButtons = document.querySelectorAll(".operator");
+operatorButtons.forEach( (operatorButton) => {
+    operatorButton.addEventListener("click", (e) => {
+        symbol = e.target.getAttribute("data-value")
+        if (characters.length === 0) {
+            clearAll();
+            updateResultDisplay("SYNTAX ERROR, HIT CLEAR");
+        } else if (OPERATORS.includes(characters.toString())) {
```

Comment:
Might be worthwhile extracting these boolean expressions out to semantically descriptive variables:

```js
const bufferContainsOperator = OPERATORS.includes(characters.toString()))
```

Reply:
I'm a bit confused about resolving the .includes code comments since there are 3 separate comments on similar code patterns. Let's talk about these.

(Lines of code 31-32, 72-74, and 94-96)